ext/pdo_sqlite/sqlite_driver.c

```diff
@@ -255,7 +255,22 @@ static bool sqlite_handle_begin(pdo_dbh_t *dbh)
 {
 	pdo_sqlite_db_handle *H = (pdo_sqlite_db_handle *)dbh->driver_data;
 
-	if (sqlite3_exec(H->db, "BEGIN", NULL, NULL, NULL) != SQLITE_OK) {
+	char *begin_statement;
+	switch (H->transaction_mode) {
+		case 0:
+		case PDO_SQLITE_ATTR_TRANSACTION_MODE_DEFERRED:
+			begin_statement = "BEGIN DEFERRED TRANSACTION";
+			break;
```

Comment:
The case `0` is when no transaction mode has been set, so we use the default (deferred).

Is it a safe assumption that `driver_data` is always zero-initialized? In the code I could find, I saw that `driver_data` is allocated with `pecalloc()`.